addons/web_editor/static/src/scss/web_editor.common.scss

```diff
@@ -935,3 +940,73 @@ section, .oe_img_bg, [data-oe-shape-data] {
     z-index: 1051;
     background-color: white;
 }
+
+// Background shapes
+@function compute-shape-url-params($colors, $color-to-cc-bg-map) {
+    $url-params: '';
+    @each $i in $colors {
+        $mapped-color: map-get($color-to-cc-bg-map, $i);
+        // %23 is the url-encoded form of '#'
+        $color: str-replace("#{o-color($mapped-color)}", '#', '%23');
+        $url-params: '#{$url-params}&c#{$i}=#{$color}';
+    }
+    @return $url-params;
+}
+
+.o_we_shape {
+    @extend %o-we-background-layer;
```

Comment:
The code was moved in common because the previews of the background shapes also use this code. (At least the definition of the url).

For the commit message, I will try to improve it before the next rebase.